service/src/test/java/com/myhome/services/unit/AuthenticationSDJpaServiceTest.java

```diff
@@ -0,0 +1,131 @@
+package com.myhome.services.unit;
+
+import com.myhome.controllers.dto.UserDto;
+import com.myhome.controllers.exceptions.CredentialsIncorrectException;
+import com.myhome.controllers.exceptions.UserNotFoundException;
+import com.myhome.domain.AuthenticationData;
+import com.myhome.model.LoginRequest;
+import com.myhome.security.jwt.AppJwt;
+import com.myhome.security.jwt.AppJwtEncoderDecoder;
+import com.myhome.services.springdatajpa.AuthenticationSDJpaService;
+import com.myhome.services.springdatajpa.UserSDJpaService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+public class AuthenticationSDJpaServiceTest {
+
+  private final String USER_ID = "test-user-id";
+  private final String USERNAME = "test-user-name";
+  private final String USER_EMAIL = "test-user-email";
+  private final String USER_PASSWORD = "test-user-password";
+  private final String REQUEST_PASSWORD = "test-request-password";
+  private final Duration TOKEN_LIFETIME = Duration.ofDays(1);
+  private final String SECRET = "secret";
+
+  @Mock
+  private UserSDJpaService userSDJpaService;
+  @Mock
+  private AppJwtEncoderDecoder appJwtEncoderDecoder;
+  @Mock
+  private PasswordEncoder passwordEncoder;
+  @InjectMocks
+  private AuthenticationSDJpaService authenticationSDJpaService;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  void loginSuccess(){
+    // given
+    LoginRequest request = getDefaultLoginRequest();
+    UserDto userDto = getDefaultUserDtoRequest();
+    AppJwt appJwt = getDefaultJwtToken(userDto);
+    String encodedJwt = appJwtEncoderDecoder.encode(appJwt, SECRET);
+    ReflectionTestUtils.setField(authenticationSDJpaService, "tokenExpirationTime", Duration.ofDays(1));
```

Comment:
Using reflection is a bad practice.
We can make `AllArgsConstructor` annotation above `AuthenticationSDJpaService` or create all args constructor by hand and use there the annotations:
```
public AuthenticationSDJpaService(@Value("${token.expiration_time}") Duration tokenExpirationTime,
                                    @Value("${token.secret}") String tokenSecret,
                                    UserSDJpaService userSDJpaService,
                                    AppJwtEncoderDecoder appJwtEncoderDecoder,
                                    PasswordEncoder passwordEncoder) {
    this.tokenExpirationTime = tokenExpirationTime;
    this.tokenSecret = tokenSecret;
    this.userSDJpaService = userSDJpaService;
    this.appJwtEncoderDecoder = appJwtEncoderDecoder;
    this.passwordEncoder = passwordEncoder;
  }
```

That will give you a way to not inject properties with reflection :)